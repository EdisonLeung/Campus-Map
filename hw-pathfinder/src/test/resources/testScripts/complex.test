# tests min pathfinding on a complex graph

# Create Graph
CreateGraph complex

# Add nodes
AddNode complex a
AddNode complex b
AddNode complex c
AddNode complex d
AddNode complex e
AddNode complex f

# Add Edges
AddEdge complex a b 1.0
AddEdge complex a e 5.0
AddEdge complex a d 3.0
AddEdge complex b c 4.0
AddEdge complex b f 2.0
AddEdge complex b e 3.0
AddEdge complex d a 2.0
AddEdge complex d e 2.0
AddEdge complex e d 4.0
AddEdge complex f c 1.0
AddEdge complex f e 3.0

# test pathfinding
FindPath complex a c

FindPath complex a d

FindPath complex a e

FindPath complex b a

FindPath complex f a

FindPath complex d f

FindPath complex e a