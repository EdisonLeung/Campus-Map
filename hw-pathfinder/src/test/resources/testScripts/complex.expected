# tests min pathfinding on a complex graph

# Create Graph
created graph complex

# Add nodes
added node a to complex
added node b to complex
added node c to complex
added node d to complex
added node e to complex
added node f to complex

# Add Edges
added edge 1.000 from a to b in complex
added edge 5.000 from a to e in complex
added edge 3.000 from a to d in complex
added edge 4.000 from b to c in complex
added edge 2.000 from b to f in complex
added edge 3.000 from b to e in complex
added edge 2.000 from d to a in complex
added edge 2.000 from d to e in complex
added edge 4.000 from e to d in complex
added edge 1.000 from f to c in complex
added edge 3.000 from f to e in complex

# test pathfinding
path from a to c:
a to b with weight 1.000
b to f with weight 2.000
f to c with weight 1.000
total cost: 4.000

path from a to d:
a to d with weight 3.000
total cost: 3.000

path from a to e:
a to b with weight 1.000
b to e with weight 3.000
total cost: 4.000

path from b to a:
b to e with weight 3.000
e to d with weight 4.000
d to a with weight 2.000
total cost: 9.000

path from f to a:
f to e with weight 3.000
e to d with weight 4.000
d to a with weight 2.000
total cost: 9.000

path from d to f:
d to a with weight 2.000
a to b with weight 1.000
b to f with weight 2.000
total cost: 5.000

path from e to a:
e to d with weight 4.000
d to a with weight 2.000
total cost: 6.000